Assets/HoloToolkit/Utilities/Scripts/PropertyChangedEventArgsEx.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.ComponentModel;
+using System.Linq.Expressions;
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    public static class PropertyChangedEventArgsEx
+    {
+        public static PropertyChangedEventArgsEx<TProperty> Create<TProperty>(string propertyName, TProperty oldValue, TProperty newValue)
+        {
+            return new PropertyChangedEventArgsEx<TProperty>(propertyName, oldValue, newValue);
+        }
+
+        public static PropertyChangedEventArgsEx<TProperty> Create<TProperty>(Expression<Func<TProperty>> memberGetter, TProperty oldValue, TProperty newValue)
+        {
+            return new PropertyChangedEventArgsEx<TProperty>(memberGetter, oldValue, newValue);
+        }
+    }
+
+    [Serializable]
+    public class PropertyChangedEventArgsEx<TProperty> :
+        PropertyChangedEventArgs
+    {
+        public TProperty OldValue { get; private set; }
+        public TProperty NewValue { get; private set; }
+
+        public PropertyChangedEventArgsEx(string inPropertyName, TProperty inOldValue, TProperty inNewValue) :
+            base(inPropertyName)
+        {
+            OldValue = inOldValue;
+            NewValue = inNewValue;
+        }
+
+        public PropertyChangedEventArgsEx(Expression<Func<TProperty>> memberGetter, TProperty inOldValue, TProperty inNewValue) :
+            this(GetMemberName(memberGetter), inOldValue, inNewValue)
+        {
+            // Nothing.
+        }
```

Comment:
No, the // Nothing is fine.

I was more interested in the reasoning behind the empty method.